test/parallel/test-fs-promises-file-handle-op-errors.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+// Flags: --expose-internals
+
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+
+// The following tests validate aggregate errors are thrown correctly
+// when both an operation and close throw.
+
+const fs = require('fs');
+const path = require('path');
+const {
+  readFile,
+  writeFile,
+  truncate,
+  lchmod,
+} = fs.promises;
+const {
+  FileHandle,
+} = require('internal/fs/promises');
+
+const assert = require('assert');
+const originalFd = Object.getOwnPropertyDescriptor(FileHandle.prototype, 'fd');
+
+let count = 0;
+async function createFile() {
+  const filePath = path.join(tmpdir.path, `op_errors_${++count}.txt`);
+  await writeFile(filePath, 'content');
+  return filePath;
+}
+
+async function checkAggregateError(op) {
+  try {
+    const filePath = await createFile();
+    Object.defineProperty(FileHandle.prototype, 'fd', {
+      get: function() {
+        // Close is set by using a setter,
+        // so it needs to be set on the instance.
+        const originalClose = this.close;
+        this.close = common.mustCall(function(...args) {
+          return originalClose.apply(this, args);
+        });
```

Comment:
Could this be simplified?
```suggestion
        this.close = common.mustCall(this.close);
```
or
```suggestion
        const originalClose = this.close;
        this.close = common.mustCall(originalClose);
```